reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/metrics/Metrics.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.metrics;
+
+import org.json.JSONObject;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.github.wnameless.json.unflattener.JsonUnflattener;
+
+public class Metrics {
+
+    private static Metrics metrics = new Metrics();
+    private ConcurrentHashMap<String, Metric<?>> registeredMetricsByName = new ConcurrentHashMap<>();
+    private final Clock clock = Clock.systemDefaultZone();
+
+    private Metrics() {
+    }
+
+    public void registerDefaultMetrics() {
+        for (MetricName metricName : MetricName.values()) {
+            registerMetric(MetricFactory.createMetric(metricName));
+        }
+    }
+
+    public void registerMetric(Metric<?> metric) {
+        registeredMetricsByName.put(metric.getName(), metric);
+    }
+
+    public void unregisterMetric(String name) {
```

Comment:
what is the use of this?

Reply:
Used in unit testing.

Reply:
Removed this class and associated unit tests as part of refactoring.